mindsdb/integrations/handlers/postgres_handler/postgres_handler.py

```diff
@@ -256,3 +263,87 @@ def get_columns(self, table_name: str) -> Response:
                 table_name = '{table_name}'
         """
         return self.native_query(query)
+
+    def subscribe(self, stop_event, callback, table_name, columns=None, **kwargs):
+
+        config = self._make_connection_args()
+        config['autocommit'] = True
+
+        conn = psycopg.connect(autocommit=True, connect_timeout=10, **config)
+
+        # create db trigger
+        trigger_name = f'mdb_notify_{table_name}'
+
+        before, after = '', ''
+
+        if columns:
+            # check column exist
+            conn.execute(f'select {",".join(columns)} from {table_name} limit 0')
+
+            columns = set(columns)
+            trigger_name += '_' + '_'.join(columns)
+
+            news, olds = [], []
+            for column in columns:
+                news.append(f'NEW.{column}')
+                olds.append(f'OLD.{column}')
+
+            before = f'IF ({", ".join(news)}) IS DISTINCT FROM ({", ".join(olds)}) then\n'
+            after = '\nEND IF;'
+        else:
+            columns = set()
+
+        func_code = f'''
+             CREATE OR REPLACE FUNCTION {trigger_name}()
+               RETURNS trigger AS $$
+             DECLARE
+             BEGIN
+               {before}
+               PERFORM pg_notify( '{trigger_name}', row_to_json(NEW)::text);
+               {after}
+               RETURN NEW;
+             END;
+             $$ LANGUAGE plpgsql;
+         '''
+        conn.execute(func_code)
+
+        # for after update - new and old have the same values
+        conn.execute(f'''
+             CREATE OR REPLACE TRIGGER {trigger_name}
+               BEFORE INSERT OR UPDATE ON {table_name}
+               FOR EACH ROW
+               EXECUTE PROCEDURE {trigger_name}();
+        ''')
+        conn.commit()
+
+        # start listen
+        conn.execute(f"LISTEN {trigger_name};")
+
+        def process_event(event):
+            try:
+                row = json.loads(event.payload)
+            except json.JSONDecoder:
+                return
+
+            # check column in input data
+            if not columns or columns.intersection(row.keys()):
+                callback(row)
+
+        try:
+            while True:
+                if stop_event.is_set():
+                    # exit trigger
+                    return
+
+                conn.add_notify_handler(process_event)
+                # trigger getting updates
+                conn.execute("SELECT 1").fetchone()
```

Comment:
spam db with query each 0.5s looks badly. Let put here comment with reminder to rewrite it to generator when psycopg 3.2.2 will be released (they added timeout arg there)

Reply:
changed timeout to 1s. 'select 1' is not heavy operation. added line on psycopg example of this approach
in future we could modify trigger to process batch operation: all rows changed since last call (1s) will be processed as one dataframe.